### ISSUE TITLE
Rename "labels" key in the slice to be "requests"

### ISSUE THREAD

**Issue by jmeas** [jmeas/redux-resource#161]
I think it makes sense for the `labels` key to be `requests` in the state tree. This is a breaking change, but I think it makes onboarding to the lib a little less weird. That section of the state tree really is explicitly all about requests, which is more familiar than 'label.'

Currently, we have:

```js
{
  resources: {},
  meta: {},
  labels: {}
}
```

This would make it so we have:

```js
{
  resources: {},
  meta: {},
  requests: {}
}
```

What do folks think?

//cc @marlonpp @tbranyen @sprjr

**Comment by sprjr** [comment]
Sounds good to me. I haven't really used this or labels yet, so these changes don't churn much on my end. Eventually when things settle down I'll try to carve out some time to get this updated.

I'm not sure how onboarding is reduced though, it's still a concept to explain a bit - maybe `labeledRequests` is more informative for what you're trying to convey?

**Comment by tbranyen** [comment]
Old way seems fine, might not be worth changing? I could see cluttering a slice.

**Comment by jmeas** [comment]
> I'm not sure how onboarding is reduced though, it's still a concept to explain a bit - maybe `labeledRequests` is more informative for what you're trying to convey?

Yeah, it doesn't completely get rid of the onboarding. I was thinking it makes the API a wee bit more expressive, which I think is where it would have a positive impact on learnin'. "resources" and "requests" are the two _things_ that this library manages for ya, so I like how in this system two of the slice names reflect those ideas.

`labeledRequests` is worth considering, too...

> I could see this cluttering a slice.

How so? The amount of data in the slice wouldn't change.

**Comment by tbranyen** [comment]
Top level keys wouldn't change? If you had 50 labels, there'd be 50 top level keys in your slice right?

**Comment by tbranyen** [comment]
Top level keys wouldn't change? If you had 50 labels, there'd be 50 top level keys in your slice right?

**Comment by jmeas** [comment]
> Top level keys wouldn't change?

✔️ Yup 

>  If you had 50 labels, there'd be 50 top level keys in your slice right?

Nah. Right now, 50 labeled requests are nested under the `labels` key. I'm saying they'd be nested under the `requests` key.

Currently:

```js
{
  labels: {
    create: { ... },
    search: { ... },
    input: { ... }
  }
}
```

Proposed:

```js
{
  requests: {
    create: { ... },
    search: { ... },
    input: { ... }
  }
}
```

The reason being that the things within this object reference a request. The label is just the tool we're using to track the request. Does that make any sense or am I cray? :)

**Comment by marlonpp** [comment]
I admit that at first I didn't understand the `labels` concept and `requests` would have made it easier.

**Comment by sprjr** [comment]
They aren't just requests though, because those also happen as a side effect of the normal-path action creators.

I think it's ok to change the name, but would maybe suggest namedRequests, or labeledRequests to help - am ok with just requests too though!

**Comment by jmeas** [comment]
> I admit that at first I didn't understand the labels concept and requests would have made it easier.

Thanks for the input, @marlonpp ! I was thinking this might be the case... 🙂 

> They aren't just requests though, because those also happen as a side effect of the normal-path action creators.

I think I understand your worry: because not _every_ request would appear in here, you're concerned that this could cause confusion. Is that right?

If so, I completely agree with you. This is _definitely_ something that we need to be mindful of.

I have a plan: calling it `requests` allows me to rewrite the documentation in such a way that I think this problem could be avoided. I think it's worth a try. If I write up some docs, would you want to take a look to see if you think it gets around the issue?

Thanks for the input, everyone!

**Comment by tbranyen** [comment]
Personally I'd prefer less changes to the API and saving it until like a v2? It does cause cognitive overhead.

**Comment by jmeas** [comment]
For sure. If I make this change, it won't be for some time.

**Comment by tbranyen** [comment]
:+1:

**Comment by jmeas** [comment]
One concern I have with this is that the key `label` in the action is better than `request`. And calling that `requestLabel`, but then putting them under the `requests` key in the slice seems like it could cause some confusion.

I think I should either:

1. keep it as `label`
2. use `requestLabel` in both the action and the slice

What do folks think?

**Comment by jmeas** [comment]
I'm thinking I'll make the action attribute `request`. So it'll look like:

```
{
  type: "READ_RESOURCES",
  request: 'getSearchResults'
}
```

**Comment by jmeas** [comment]
Resolved in #195
